netbox_branching/merge_strategies/squash.py

```diff
@@ -0,0 +1,790 @@
+"""
+Squash merge strategy implementation with functions for collapsing and ordering ObjectChanges.
+"""
+import logging
+
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import DEFAULT_DB_ALIAS, models
+
+from core.choices import ObjectChangeActionChoices
+from netbox.context_managers import event_tracking
+from netbox_branching.utilities import update_object
+from utilities.data import shallow_compare_dict
+from utilities.exceptions import AbortTransaction
+from utilities.serialization import deserialize_object
+
+from .strategy import MergeStrategy
+
+
+__all__ = (
+    'SquashMergeStrategy',
+)
+
+
+class CollapsedChange:
+    """
+    Represents a collapsed set of ObjectChanges for a single object.
+    """
+    def __init__(self, key, model_class):
+        self.key = key  # (content_type_id, object_id)
+        self.model_class = model_class
+        self.changes = []  # List of ObjectChange instances, ordered by time
+        self.final_action = None  # 'create', 'update', 'delete', or 'skip'
+        self.prechange_data = None
+        self.postchange_data = None
```

Comment:
Can we initialize these to empty dicts to avoid having to do it under `collapse()`?
```suggestion
        self.prechange_data = {}
        self.postchange_data = {}
```